### PR TITLE
[WIP] Support promtail reload config

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <764524258@qq.com>

### CODE REVIEW

**Comment by daixiang0** [comment]
Use fsnotify to watch config changes, it is only a start.

**Comment by daixiang0** [comment]
Fix #16

**Comment by davkal** [comment]
What do you think about the approach @gouthamve ?

**Comment by daixiang0** [comment]
@davkal @gouthamve there is some way to support reload i think:
- watch config file
- watch config content
- only support in k8s, use Reloader plugin

**Comment by steven-sheehy** [comment]
Reload is also supported automatically if you exclusively use helm to manage the configmap and put the below annotation on the daemonset.yaml:

`checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}`

**Comment by daixiang0** [comment]
I know it, but it limits deploy way indeed.

**Comment by daixiang0** [comment]
hi, @gouthamve mind have a look?

File: pkg/promtail/client/client.go [hunk 1]
```diff
@@ -161,6 +161,10 @@ func (c *Client) sendBatch(batch map[model.Fingerprint]*logproto.Stream) {
 	encodedBytes.Add(bufBytes)
 
 	ctx := context.Background()
+	level.Info(c.logger).Log("msg", "print backoff config", "value", c.cfg.BackoffConfig.MinBackoff)
```
Comment (gouthamve) [inline pkg/promtail/client/client.go]: These can be removed?

Comment (daixiang0) [reply pkg/promtail/client/client.go]: Yep, i will change to `/reload`.

**Comment by daixiang0** [comment]
But when i read more details about Prometheus, [it](https://github.com/prometheus/prometheus/blob/master/discovery/file/file.go) still call `fsnotify`.

**Comment by gouthamve** [comment]
Yes. But that is for fileSD and not the actual config file.

When it is picking up targets from file it automatically reloads new
targets from the file but the actual config reload happens only through
manual operator input. This is required as we don't want to reload the
config while we are in the middle of changing it, etc.

**Comment by daixiang0** [comment]
Get it, i will do like it.

**Pull request closed without merging**